custom_components/waste_collection_schedule/waste_collection_schedule/source/boden_se.py

```diff
@@ -0,0 +1,106 @@
+from datetime import datetime
+import json
+import logging
+
+import requests
+from waste_collection_schedule import Collection  # type: ignore[attr-defined]
+
+TITLE = "Boden Avfall och Återbruk"
+DESCRIPTION = "Source for Boden Avfall och Återbruk waste collection."
+URL = "https://www.boden.se"
+TEST_CASES = {
+    "Bodens Kommun": {"street_address": "KYRKGATAN 24, Boden"},
+    "Gymnasiet": {"street_address": "IDROTTSGATAN 4 , BJÖRKNÄSGYMNASIET, Boden"},
+}
+COUNTRY = "se"
+_LOGGER = logging.getLogger(__name__)
+
+# This is based on the ssam_se source, just changed to work with boden.se
+class Source:
+    def __init__(self, street_address):
+        self._street_address = street_address
+
+    def fetch(self):
+        params = {"searchText": self._street_address}
+        response = requests.post(
+            "https://edpmobile.boden.se/FutureWeb/SimpleWastePickup/SearchAdress",
+            params=params,
+            timeout=30,
+        )
+
+        address_data = json.loads(response.text)
+        address = None
+
+        if address_data and len(address_data) > 0:
+            address = address_data["Buildings"][0]
+
+        if not address:
+            return []
```

Comment:
please raise an Exception as this will be shown to the user. You implementation will just return an empty calendar without any further information
e.g.:
```python
if not address:
    raise Exception("Address not found")
```